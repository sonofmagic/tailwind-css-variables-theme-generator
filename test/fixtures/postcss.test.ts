import postcss from 'postcss'
import tailwindcss from 'tailwindcss'

describe('postcss plugin', () => {
  async function getCss (content: string[]) {
    const processor = postcss([
      tailwindcss({
        content: content.map((x) => {
          return {
            raw: x
          }
        }),
        plugins: []
      })
    ])
    return await processor.process('@tailwind utilities;', {
      from: 'index.css',
      to: 'index.css'
    })
  }

  it('base', async () => {
    const res = await getCss([
      `<div class="ring-white">
    <p>I have a white ring.</p>
</div>`
    ])
    expect(res.css.toString()).toMatchSnapshot()
  })
})
