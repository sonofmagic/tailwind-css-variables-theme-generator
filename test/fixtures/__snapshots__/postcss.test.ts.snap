// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss plugin base 1`] = `
".ring {
    --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
    --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);
    box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow, 0 0 #0000)
}
.ring-white {
    --tw-ring-opacity: 1;
    --tw-ring-color: rgb(255 255 255 / var(--tw-ring-opacity))
}"
`;
